scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModel.scala

```diff
@@ -315,6 +316,15 @@ object H2OMOJOCache extends H2OMOJOBaseCache[EasyPredictModelWrapper, H2OMOJOMod
     }
   }
 
+  private def canGenerateStageResults(model: GenModel): Boolean = {
+    model match {
+      case _: SharedTreeMojoModel => true
```

Comment:
Can we also check here whether the model was xgboost and fail early?

Reply:
IMHO, It will fail on XGBoost here since `XGBoostMojoModel` since doesn't inherit from `SharedTreeMojoModel`.

Reply:
Thanks, didn't know that!